test/eredis_tcp_SUITE.erl

```diff
@@ -327,6 +341,132 @@ t_reconnect(Config) when is_list(Config) ->
+                                ok = gen_tcp:close(ClientSocket)
+                        end),
+    process_flag(trap_exit, true),
+    Res = eredis:start_link("127.0.0.1", Port, [{database, 2},
+                                                {reconnect_sleep, no_reconnect}]),
+    ?assertMatch({error, {select_error, closed}}, Res),
+    IsDead = receive {'EXIT', _, _} -> died
+             after 2000 -> still_alive end,
+    process_flag(trap_exit, false),
+    ?assertEqual(died, IsDead),
+    eredis_test_utils:stop_server(Pid).
+
+t_send_error(Config) when is_list(Config) ->
+    %% Verify handling of send errors by using the send_timout_close option.
+    %% Connect an eredis client to a server and fill the clients sockets sendbuffer
+    %% until the socket is full. Then let eredis handle a query which will
+    %% trigger a send error (and reconnection attempts)
+    {ok, Pid, Port} = eredis_test_utils:start_server(
+                        fun(ClientSocket) ->
+                                inet:setopts(ClientSocket, [{recbuf, 512}])
+                        end),
+    {ok, C} = eredis:start_link("127.0.0.1", Port, [{reconnect_sleep, 100},
+                                                    {socket_options, [{send_timeout, 250},
+                                                                      {send_timeout_close, true},
+                                                                      {sndbuf, 512}]}]),
+    eredis_test_utils:await_connect(Pid),
+    [Socket] = get_tcp_ports(C),                %% Get eredis client socket
+    {error, timeout} = send_data_loop(Socket),  %% Fill the socket
+    ?assertEqual({error, enotconn}, eredis:q(C, ["GET", foo])),
+    timer:sleep(200),
+    [Socket2] = get_tcp_ports(C),               %% Get new eredis client socket
+    ?assertNotEqual(Socket, Socket2),           %% Reconnection attempts with new socket
+    eredis_test_utils:stop_server(Pid).
+
+send_data_loop(Socket) ->
+    Data = << <<0>> || _ <- lists:seq(1, 1024) >>,
+    case gen_tcp:send(Socket, Data) of
+        ok -> send_data_loop(Socket);
+        Error -> Error
+    end.
+
+t_ignore_old_socket(Config) when is_list(Config) ->
+    %% Ignore messages and errors for an old socket
+    {ok, C} = eredis:start_link("127.0.0.1", ?PORT, []),
+    [Socket] = get_tcp_ports(C), %% Get eredis client socket
+    C ! {tcp, old_socket, data},
+    timer:sleep(200),
+    [Socket] = get_tcp_ports(C), %% Still same socket port, i.e ignored
+    C ! {tcp_error, old_socket, reason},
+    timer:sleep(200),
+    [Socket] = get_tcp_ports(C),
+    C ! {tcp_closed, old_socket},
+    timer:sleep(200),
+    [Socket] = get_tcp_ports(C),
+    C ! {send_error, old_socket, reason},
+    timer:sleep(200),
+    [Socket] = get_tcp_ports(C).
+
+t_unhandled_message(Config) when is_list(Config) ->
+    %% Verify that when receiving an unhandled message the client is stopped/killed
```

Comment:
I hope it's not killed by system messages though... Maybe `gen_server` is handling them without even calling `handle_info` for these...? I assume it does. Functions like `sys:get_state(Pid)` are implemented using sys messages.

IMO, it's fine for handle_cast and handle_call to not handle unexpected stuff, but for handle_info, it would be more safe to ignore them and just return noreply.

Reply:
Yes, I dont know the reason for it ..maybe it better to just remove the handle function and let the default take it: 
"Module:handle_info(Info, State) -> Result
The gen_server module provides a default implementation of this function that logs about the unexpected Info message, drops it and returns {noreply, State}."

Reply:
If we remove handle_info, we can't handle tcp_closed or even tcp data... Not really an option. :-)

Reply:
yes, noticed that when starting to remove it :)  changed it to a noreply